### PR TITLE
implemented modPromptButton

### PR DESCRIPTION
**Pull request by TAImatem**
doesn't really depend on IsSelected, but next PR will depend on both ;)

### CODE REVIEW

**Comment by TAImatem** [comment]
Title is replaced with Prompt, kinda incompatible

**Comment by TAImatem** [comment]
Inhereting from ModTitleButton might make it work shoddily 😝

File: OWML.Common/Menus/IModPromptButton.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+        new IModPromptButton Copy();
+        IModPromptButton Copy(string title);
+        IModPromptButton Copy(string title, int index);
+
+        IModPromptButton Duplicate(string title);
+        IModPromptButton Duplicate(string title, int index);
+
+        IModPromptButton Replace(string title);
+        IModPromptButton Replace(string title, int index);
```
Comment (amazingalek) [inline OWML.Common/Menus/IModPromptButton.cs:16]: I suggest leaving this out, until/unless you need them. In many cases I don't think we end up using these, and many UI elements don't have/need them. We have the base stuff anyways

File: OWML.ModHelper.Menus/ModPromptButton.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            {
+                _prompt = value;
+                _hotkeyButton.SetPrompt(value);
+            }
+        }
+
+        public ModPromptButton(Button button, IModMenu menu) : base(button, menu)
+        {
+            _hotkeyButton = Button.GetComponent<ButtonWithHotkeyImageElement>();
+            _textID = Button.GetComponentInChildren<LocalizedText>(true).GetValue<UITextType>("_textID");
```
Comment (amazingalek) [inline OWML.ModHelper.Menus/ModPromptButton.cs:31]: This doesn't have to happen if the null check below fails, so move it below the null check?

File: OWML.Common/Menus/IModPromptButton.cs (hunk 1 above)
Comment (TAImatem) [reply OWML.Common/Menus/IModPromptButton.cs:16]: yeah, makes sense

File: OWML.ModHelper.Menus/ModPromptButton.cs (hunk 2 above)
Comment (TAImatem) [reply OWML.ModHelper.Menus/ModPromptButton.cs:31]: K

File: OWML.Common/Menus/IModPromptButton.cs (hunk 1 above)
Comment (TAImatem) [reply OWML.Common/Menus/IModPromptButton.cs:16]: since it's basically only intended for sugarcoating existing buttons

Comment (TAImatem) [reply OWML.Common/Menus/IModPromptButton.cs:16]: and if someone wants to copy it (s)he would probably also want to set new Prompt

File: OWML.ModHelper.Menus/ModPromptButton.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        private ScreenPrompt _prompt;
+        private UITextType _textID;
+        private readonly ButtonWithHotkeyImageElement _hotkeyButton;
+
+        public string DefaultTitle
+        {
+            get => UITextLibrary.GetString(_textID);
+        }
+        public ScreenPrompt Prompt
```
Comment (amazingalek) [inline OWML.ModHelper.Menus/ModPromptButton.cs:18]: newline

Comment (TAImatem) [reply OWML.ModHelper.Menus/ModPromptButton.cs:17]: Do I actually need newline between getters? pretty sure it's missing in other classes 😛

Comment (TAImatem) [reply OWML.ModHelper.Menus/ModPromptButton.cs:17]: DefaultTitle can be turned into one-liner though

Comment (amazingalek) [reply OWML.ModHelper.Menus/ModPromptButton.cs]: I've not seen any :O

Comment (amazingalek) [reply OWML.ModHelper.Menus/ModPromptButton.cs]: Actually, single line getters can be close to each other, but not multiline ones ;)

Comment (amazingalek) [reply OWML.ModHelper.Menus/ModPromptButton.cs]: Eh, it's fine, just resolve the conflicts and we're good ;)

File: OWML.Common/Menus/IModInputBase.cs [hunk 4]
```diff
@@ -1,4 +1,4 @@
-﻿using UnityEngine;
+using UnityEngine;
```
Comment (amazingalek) [inline OWML.Common/Menus/IModInputBase.cs:1]: Hmmm? :P

**Pull request merged**